src/Illuminate/Routing/RouteUri.php

```diff
@@ -4,6 +4,18 @@
 
 class RouteUri
 {
+    /**
+     * Registered field types and their associated regular expressions.
+     *
+     * @var array
+     */
+    protected static $typeExpressions = [
+        'int' => '[0-9]+',
```

Comment:
do we want to allow leading zeros?

Reply:
I thought about that, but I think leading zeros allows for more common URL patterns without breaking much. We could potentially use `number` to match `whereNumber()` and then make `int` be more strict. But I personally like `int` because it matches my PHP:

```php

Route::get('/events/{int id}', EventController::class);

class EventController
{
  public function __invoke(int $id)
  {
    // $id is always a valid int
  }
}
```

Reply:
Allowing leading zeros leads to non-normalized URLs.

Reply:
I guess people can do a custom pattern, and what you have here best matches the current behaviour for looking up models with integer primary keys.